### PR TITLE
(feat) allow customization for the speed of the cat :3

### PR DESCRIPTION
**Pull request by not-varram**

### CODE REVIEW

**Comment by not-varram** [comment]
this has been bugging me for way too long hahaha (and I don't feel like manually making a new dev build for everytime this thing gets updated)

File: src/plugins/oneko/index.ts [hunk 1]
```diff
@@ -16,18 +16,44 @@
...
 
+import { Settings } from "@api/Settings";
 import { Devs } from "@utils/constants";
-import definePlugin from "@utils/types";
+import definePlugin, { OptionType } from "@utils/types";
 
 export default definePlugin({
     name: "oneko",
-    description: "cat follow mouse (real)",
+    description: "cat follow mouse (real) x2 speed",
```
Comment (lewisakura) [inline src/plugins/oneko/index.ts]: ?

File: src/plugins/oneko/index.ts [hunk 2]
```diff
@@ -16,18 +16,44 @@
...
+                        .then(x => x.text())
+                        .then(s => s.replace("const nekoSpeed = 10;", `const nekoSpeed = ${Settings.plugins.oneko.randomisedLength};`))
+                        .then(s => s.replace("./oneko.gif", "https://raw.githubusercontent.com/adryd325/oneko.js/14bab15a755d0e35cd4ae19c931d96d306f99f42/oneko.gif")
+                            .replace("(isReducedMotion)", "(false)"))
+                        .then(eval);
+                }
+                return;
+            }
+        },
+    },
```
Comment (lewisakura) [inline src/plugins/oneko/index.ts:51]: Use `definePluginSettings`, the options API is deprecated.

File: src/plugins/oneko/index.ts [hunk 3]
```diff
@@ -16,18 +16,44 @@
...
+            onChange: () => {
+                if (Settings.plugins.oneko.enabled) {
+                    document.getElementById("oneko")?.remove();
+                    fetch("https://raw.githubusercontent.com/adryd325/oneko.js/8fa8a1864aa71cd7a794d58bc139e755e96a236c/oneko.js")
+                        .then(x => x.text())
+                        .then(s => s.replace("const nekoSpeed = 10;", `const nekoSpeed = ${Settings.plugins.oneko.randomisedLength};`))
+                        .then(s => s.replace("./oneko.gif", "https://raw.githubusercontent.com/adryd325/oneko.js/14bab15a755d0e35cd4ae19c931d96d306f99f42/oneko.gif")
+                            .replace("(isReducedMotion)", "(false)"))
+                        .then(eval);
+                }
```
Comment (lewisakura) [inline src/plugins/oneko/index.ts:47]: Code duplication no bueno, just call `stop()` and `start()`.

File: src/plugins/oneko/index.ts [hunk 4]
```diff
@@ -16,18 +16,44 @@
...
+                return;
+            }
+        },
+    },
 
     start() {
         fetch("https://raw.githubusercontent.com/adryd325/oneko.js/8fa8a1864aa71cd7a794d58bc139e755e96a236c/oneko.js")
             .then(x => x.text())
+            .then(s => s.replace("const nekoSpeed = 10;", `const nekoSpeed = ${Settings.plugins.oneko.randomisedLength};`))
```
Comment (lewisakura) [inline src/plugins/oneko/index.ts:56]: Replace with new settings API.

**Review by lewisakura — CHANGES_REQUESTED** [review]

**Comment by not-varram** [comment]
Fixed all the issues pointed out in the comments :D

**Pull request closed without merging**